### ISSUE TITLE
RangeError in webcrypto-core.padStart

### ISSUE THREAD

**Issue by Champ-Goblem** [PeculiarVentures/pvtsutils#12]
Since the version change to 1.3.3 of pvtsutils we have noticed a couple of errors that are currently causing ECDSA certificate generation to fail.

If we pin the version back to 1.3.2 the issue no longer happens.

When running locally babel complains with the following error: `The argument can't be less than 0`

```
RangeError: The argument can't be less than 0
    at module.exports (/node_modules/@babel/node/node_modules/core-js/internals/to-positive-integer.js:8:25)
    at module.exports (/node_modules/@babel/node/node_modules/core-js/internals/to-offset.js:7:16)
    at Uint8Array.set (/node_modules/@babel/node/node_modules/core-js/modules/es.typed-array.set.js:36:16)
    at Function.padStart (/node_modules/webcrypto-core/build/webcrypto-core.js:972:13)
    at Function.encodeSignature (/node_modules/webcrypto-core/build/webcrypto-core.js:940:22)
    at Function.sign (/node_modules/@peculiar/webcrypto/build/webcrypto.js:1575:54)
    at EcdsaProvider.onSign (node_modules/@peculiar/webcrypto/build/webcrypto.js:1737:25)
    at EcdsaProvider.sign (/node_modules/webcrypto-core/build/webcrypto-core.js:157:28)
    at SubtleCrypto.sign (/node_modules/webcrypto-core/build/webcrypto-core.js:1405:39)
    at Function.create /node_modules/@peculiar/x509/build/x509.cjs.js:2407:35)
```

Which is probably some extra checks babel performs. However, without babel we get the following error:

```
RangeError: offset is out of bounds
    at Uint8Array.set (<anonymous>)
    at Function.padStart (/node_modules/webcrypto-core/build/webcrypto-core.js:972:13)
    at Function.encodeSignature (/node_modules/webcrypto-core/build/webcrypto-core.js:940:22)
    at Function.sign (/node_modules/@peculiar/webcrypto/build/webcrypto.js:1575:54)
    at EcdsaProvider.onSign (/node_modules/@peculiar/webcrypto/build/webcrypto.js:1737:25)
    at EcdsaProvider.sign (/node_modules/webcrypto-core/build/webcrypto-core.js:157:28)
    at SubtleCrypto.sign (/node_modules/webcrypto-core/build/webcrypto-core.js:1405:39)
    at Function.create (/node_modules/@peculiar/x509/build/x509.cjs.js:2407:35)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
```

The following code snippet can recreate the problem:
```
const x509 = require('@peculiar/x509');
const {Crypto} = require('@peculiar/webcrypto');
const {randomBytes} = require('crypto');
const moment = require('moment');

const defaultSerialBytes = 19;
const getRandomSerialNumber = () => {
  const sn = randomBytes(defaultSerialBytes).toString('hex');
  return `01${sn}`;
};

(async () => {
  const alg = {
    name: 'ECDSA',
    namedCurve: 'P-256',
    hash: 'SHA-256',
    usages: ['sign', 'verify'],
  };

  const crypto = new Crypto();
  x509.cryptoProvider.set(crypto);
  const keys = await crypto.subtle.generateKey(alg, true, alg.usages);

  const cert = await x509.X509CertificateGenerator.createSelfSigned({
    serialNumber: getRandomSerialNumber(),
    name: `CN=test.com`,
    notBefore: moment().toDate(), 
    notAfter: moment().add(10, 'days').toDate(),
    signingAlgorithm: alg,
    keys,
    extensions: [
      new x509.BasicConstraintsExtension(
        false,
        0,
        true
      ),
      await x509.SubjectKeyIdentifierExtension.create(keys.publicKey),
    ],
  });

  const privateKey = await crypto.subtle.exportKey('pkcs8', keys.privateKey);
  const publicKey = await crypto.subtle.exportKey('spki', keys.publicKey);

  return { certificate: cert.toString('pem'), privateKey, publicKey };

})().catch(console.error);
```

**Comment by maeri** [comment]
That bug caused me 2 days of verification - please have a look asap if possible. Thanks.

**Comment by CMCDragonkai** [comment]
I'm hitting this problem as well.

It came from an update to `@peculiar/x509`. From `1.9.3` to `1.9.4`.

This pulled in an update for `pvtsutils` from `1.3.2` to `1.3.4`.

I traced this down to `pvtsutils` so I have pinned `pvtsutils` to `1.3.2` while allowing `@peculiar/x509` to update.

Using `pvtsutils` of `1.3.3` or `1.3.4` causes errors in ECDSA key generation and RSA key generation, but not Ed25519 key generation.

I can provide a replication soon.

**Comment by microshine** [comment]
I've released a new build of `pvtsutils@1.3.5`. This update addresses an issue introduced in version `1.3.3`. The problem was that the `BufferSourceConverter.toView` method used `instanceof` for checking, which prevented `Buffer` from being cast to `Uint8Array`. Modules using this method followed by a call to `slice` for `Uint8Array` received a different result, as `Buffer.slice` returns a different outcome compared to `Uint8Array.slice`. I reverted to the check from version `1.3.2`, and the issue has been resolved.

Please try the latest version and close the issue if it's solved

**Comment by CMCDragonkai** [comment]
Great that fixed it! Thanks. It works for ECDSA and RSA and Ed25519 now.
